Release notes 7.0.37-20 (#1137)
percona/psmdb-docs: docs/release_notes/7.0.37-20.md

---
version: 7.0.37-20
---

# Percona Server for MongoDB {{ page.meta.version }} ({{date.7_0_37}})

[Installation](../install/index.md){.md-button}
[Upgrade from MongoDB Community](../install/upgrade-from-mongodb.md){.md-button}

Percona Server for MongoDB {{ page.meta.version }} is an enhanced, source-available, and highly-scalable database that is a fully-compatible, drop-in replacement for MongoDB Community Edition.

Percona Server for MongoDB **{{ page.meta.version }}** includes the improvements and bug fixes of:

> - [MongoDB 7.0.37 Community Edition :octicons-link-external-16:](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.37---june-11--2026){:target="_blank"}

Reviewer: List all incl. .36, and .35

Author: ok

- Supports protocols and drivers of MongoDB Community **7.0.37**.

## Upgrade recommendation

> This release contains multiple high-severity security fixes affecting all Percona Server for MongoDB 7.0.x versions. We strongly recommend upgrading to version 7.0.37 as soon as possible.

Reviewer: ```suggestion
This release contains multiple high-severity security fixes affecting all Percona Server for MongoDB 7.0.x versions. We strongly recommend upgrading to version {{ page.meta.version }} as soon as possible.
```

## Security updates: CVE fixes from upstream MongoDB

### Affected versions

These vulnerabilities affect the following versions:

- All Percona Server for MongoDB 7.0.x versions

> - MongoDB Community 7.0 versions prior to 7.0.37

Reviewer: this one is not needed - it's our release notes not upstream

- [SERVER-123951 :octicons-link-external-16:](https://jira.mongodb.org/browse/SERVER-123951){:target="_blank"} **(CVE-2026-9748):** Under specific conditions, using the internal `$_internalConvertBucketIndexStats` stage together with `$facet` can cause the `mongod` process to terminate unexpectedly.

- [SERVER-123918 :octicons-link-external-16:](https://jira.mongodb.org/browse/SERVER-123918){:target="_blank"} **(CVE-2026-9747):** A vulnerability triggered by using `fromRouter: true` together with `runtimeConstants.userRoles` can cause the `mongod` process to terminate unexpectedly.

- [SERVER-124190 :octicons-link-external-16:](https://jira.mongodb.org/browse/SERVER-124190){:target="_blank"} **(CVE-2026-9746):** A vulnerability in the use of `$changeStream`, `$_requestReshardingResumeToken`, and the exchange option can cause the `mongod` process to terminate unexpectedly. An authenticated user can trigger this behavior without requiring any special privileges.

> - [SERVER-123507 :octicons-link-external-16:](https://jira.mongodb.org/browse/SERVER-123507){:target="_blank"} **(CVE-2026-9741):** Queries using `$vectorSearch` with Queryable Encryption (QE) or Client-Side Field Level Encryption (CSFLE) could transmit encrypted field values in filter expressions as plaintext instead of ciphertext. This could expose sensitive data.

Reviewer: @sandraromanchenko is that relevant for 7.0 if we don't have $vectorSearch ?

Reviewer 2: You're right, I don't see any code in community PSMDB related to vector search. Not sure whether task was mislabeled or they keep the code for enterprise. 
@radoslawszulgo

Author: @sandraromanchenko, I will remove it then.

Reviewer 3: v7.0 codebase has some mentions of $vectorSearch however there is no commit for SERVER-123507 on the v7.0 branch
commit for SERVER-124190 exists

Author: @radoslawszulgo @igorsol, so we should remove it or keep it?

Reviewer 3: My vote:
- keep SERVER-124190
- remove SERVER-123507

### Medium severity

> - [SERVER-123370 :octicons-link-external-16:](https://jira.mongodb.org/browse/SERVER-123370){:target="_blank"} **(CVE-2026-9751):** When the `ldapQueryPassword` parameter is set using the runtime `setParameter` command, the new password value is written to `mongod.log` in plain text. This can expose sensitive credentials in log files.

Reviewer: @igorsol  is that fix relevant to us having our own implementation of LDAP?

Author: @radoslawszulgo Do we wanna keep this?

Reviewer 3: @radoslawszulgo @rasika-chivate I think we need to keep this however we can improve description.
The problem was detected with `ldapQueryPassword` but the fix is more general: it is implemented for all `setParameter` commands. Now for all `setParameter` commands:
1. all fields flagges with `isRedact` are not logged
2. unrecognized parameters are redacted in case the user made a typo like "ldapQueryPasssword"

---
version: 7.0.37-20
---

# Percona Server for MongoDB {{ page.meta.version }} ({{date.7_0_37}})

[Installation](../install/index.md){.md-button}
[Upgrade from MongoDB Community](../install/upgrade-from-mongodb.md){.md-button}

Percona Server for MongoDB {{ page.meta.version }} is an enhanced, source-available, and highly-scalable database that is a fully-compatible, drop-in replacement for MongoDB Community Edition.

Percona Server for MongoDB **{{ page.meta.version }}** includes the improvements and bug fixes of:

> - [MongoDB 7.0.37, 7.0.36, and 7.0.35 Community Edition :octicons-link-external-16:](https://www.mongodb.com/docs/manual/release-notes/7.0/#7.0.37---june-11--2026){:target="_blank"}

Reviewer: We used to link each release separately. See https://github.com/percona/psmdb-docs/blob/8.0/docs/release_notes/8.0.16-5.md

Author: Will fix it.

- Supports protocols and drivers of MongoDB Community **7.0.37**.

## Upgrade recommendation

This release contains multiple high-severity security fixes affecting all Percona Server for MongoDB 7.0.x versions. We strongly recommend upgrading to version {{ page.meta.version }} as soon as possible.

## Improvements

> - [PSMDB-2038](https://perconadev.atlassian.net/browse/PSMDB-2038): Percona Server for MongoDB now exposes LDAP `userToDN` cache statistics through the serverStatus command. These metrics provide visibility into cache utilization and effectiveness, helping administrators troubleshoot LDAP authentication latency, validate cache behavior after configuration changes, and optimize cache sizing for their workloads. The new `ldap.userToDNCache` section reports runtime information such as cache usage, hits, misses, and invalidations, making LDAP authentication performance easier to monitor and tune.

Reviewer: ```suggestion
- [PSMDB-2038](https://perconadev.atlassian.net/browse/PSMDB-2038): Percona Server for MongoDB now exposes LDAP `userToDN` cache statistics through the `serverStatus` command. These metrics provide visibility into cache utilization and effectiveness, helping administrators troubleshoot LDAP authentication latency, validate cache behavior after configuration changes, and optimize cache sizing for their workloads. The new `ldap.userToDNCache` section reports runtime information such as cache usage, hits, misses, and invalidations, making LDAP authentication performance easier to monitor and tune.
```

> ## Security updates: CVE fixes from upstream MongoDB

Reviewer: ```suggestion
## Security updates
```

- [SERVER-123918 :octicons-link-external-16:](https://jira.mongodb.org/browse/SERVER-123918){:target="_blank"} **(CVE-2026-9747):** A vulnerability triggered by using `fromRouter: true` together with `runtimeConstants.userRoles` can cause the `mongod` process to terminate unexpectedly.

- [SERVER-124190 :octicons-link-external-16:](https://jira.mongodb.org/browse/SERVER-124190){:target="_blank"} **(CVE-2026-9746):** A vulnerability in the use of `$changeStream`, `$_requestReshardingResumeToken`, and the exchange option can cause the `mongod` process to terminate unexpectedly. An authenticated user can trigger this behavior without requiring any special privileges.

### Medium severity

- [SERVER-123370 :octicons-link-external-16:](https://jira.mongodb.org/browse/SERVER-123370){:target="_blank"} **(CVE-2026-9751):** When the `ldapQueryPassword` parameter is set using the runtime `setParameter` command, the new password value is written to `mongod.log` in plain text. This can expose sensitive credentials in log files.

> ## Bugs fixed

Reviewer 3: Probably we should mention PSMDB-2071 bug fix. It eliminates rare issue which could happen during several major version upgrade. @radoslawszulgo @sandraromanchenko what do you think?

Author: @radoslawszulgo @sandraromanchenko Could you please confirm?

Reviewer: agree - but it's reported as an improvement.
